src/locationsettings.cpp

```diff
@@ -466,6 +479,12 @@ bool LocationSettings::locationEnabled() const
 void LocationSettings::setLocationEnabled(bool enabled)
 {
     Q_D(LocationSettings);
+
+    if (d->isEmergencyCallActive()) {
+        qWarning() << "Cannot allow to change location enabled status when emergency call is active";
```

Comment:
Just wondering whether we want debug logging for any attempt to enable/disable, or just for the ones that would actually lead to changes i.e. should this be withing the below "if it differs from current" conditional block?

Also, is it okay for qml property setter not to set the property? I'd guess yes, but does somebody know for sure?

Reply:
Sort of related, if there were properties like "canToggleLocationEnabled" or stte -> those could perhaps be used in (qml) ui to disable/hide those element that could lead to attempts to make such changes?

Reply:
That is a good option I think, to have a `canToggleLocation` visible upwards to UI.

Reply:
Added some with 517f7bd029dc2e9cc22e1b2323b63db32d0a3659 -> naming needs to be rethought. On Friday evening brain just shortcircuits.